vis_gen.py

```diff
@@ -0,0 +1,565 @@
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        # Read file and translate format
+        code_points = []
+        with open(file_path) as f:
+            for line in f:
+                code_point = chr(int('0x' + line.split('\n')[0][2:], 16))
+                code_points.append(code_point)
+
+        self.generate_dataset_from_list(code_points, font_styles, antialiases)
+
+
+    def generate_dataset_from_list(self, code_points, font_styles, antialiases):
+        """Render text images in the given list of code points and write to
+        out_dir.
+
+        Args:
+            code_points: list of Unicode code points to visualize
+            font_styles: list of Str, styles of the font face to visualize
+            antialiases: list of Str, antialiasing styles to visualize
+
+        Raises:
+            OSError: if specified directory cannot be created
+            TypeError: if code point format is not correct
+        """
+        # Check if out_dir exists and create if not
+        out_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        if self.__check_out_dir:
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        for font_style in font_styles:
+            self.font_style = font_style
+            print('Successfully selected font style: {}.'.format(font_style))
+            for antialias in antialiases:
+                self.antialias = antialias
+                print('Successfully selected antialiasing style: {}.'.format(
+                    antialias))
+                self.visualize_list(code_points)
+
+        self.__check_out_dir = True
+
+
+    def visualize_list(self, code_points, x=None, y=None):
+        """Render text images from start code point to end code point and write
+        to out_dir.
+
+        Args:
+            code_points: List of Str, a list of all codepoints to visualize
+            x: Int, x coordinate of the position of text, in number of pixels
+            y: Int, y coordinate of the position of text, in number of pixels
+
+        Raises:
+            OSError: if specified directory cannot be created
+            TypeError: if code point format is not correct
+        """
+        # Check if out_dir exists and create if not
+        out_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        if self.__check_out_dir:
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        # Visualize list of code points
+        print("Visualizing {} total code points.".format(len(code_points)))
+        for idx, code_point in enumerate(code_points):
+            if idx % 50 == 0:
+                print("Now writing {}st code point.".format(idx + 1))
+            self.visualize_single(code_point, False, x=x, y=y)
+        print("Finished.")
+        print("Images stored in directory {}.".format(out_dir_abs))
+
+        self.__check_out_dir = True
```

Comment:
Why do you need to "self.__check_out_dir = True" here?

Reply:
self._check_out_dir is just a flag to specify whether or not out_dir should be checked. If in the parent function, out_dir is already checked and created, the subsequent functions won't check again. I used to have performance issue with function `os.path.isdir`, so I am trying to avoid checking over and over again.

Reply:
Will add more comment about this.